features/mbedtls/mbed-crypto/platform/TARGET_PSA/mbed_lib.json

```diff
@@ -1,4 +1,4 @@
 {
     "name": "mbed-crypto",
-    "macros": ["MBEDTLS_PSA_HAS_ITS_IO"]
+    "macros": ["MBEDTLS_PSA_INJECT_ENTROPY"]
```

Comment:
To build Mbed Crypto with the entropy injection feature, you need:

* Storage support in Mbed Crypto: `MBEDTLS_PSA_CRYPTO_STORAGE_C` must be enabled. This requirement already existed in Mbed OS 5.12 (Mbed Crypto 2.17).
* ITS storage backend support in Mbed Crypto. In Mbed OS 5.12, this required `MBEDTLS_PSA_HAS_ITS_IO`. With Mbed Crypto 2.18, you don't need anything beyond `MBEDTLS_PSA_CRYPTO_STORAGE_C`, but you need to make sure that `MBEDTLS_PSA_ITS_FILE_C` is not enabled since Mbed OS comes with its own ITS storage implementation.
* Entropy file support must be enabled, i.e. `MBEDTLS_ENTROPY_NV_SEED` must be defined. This requirement was already present before.
* Default entropy sources must be disabled, i.e. `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES` must be defined. This requirement was already present before.
* `MBEDTLS_PSA_INJECT_ENTROPY` must be defined. Before, the feature was automatically enabled if both `MBEDTLS_PSA_HAS_ITS_IO` and `MBEDTLS_ENTROPY_NV_SEED` were enabled.

So I don't understand why it worked before. What caused `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES`, `MBEDTLS_ENTROPY_NV_SEED`, and for that matter `MBEDTLS_PSA_CRYPTO_STORAGE_C` and ``MBEDTLS_PSA_CRYPTO_C` to be enabled? Any idea @k-stachowiak ?

Reply:
I found that `MBEDTLS_PSA_CRYPTO_STORAGE_C` is enabled conditionally in `features/mbedtls/platform/inc/platform_mbed.h`. In this file, `#define MBEDTLS_PSA_CRYPTO_STORAGE_ITS_C` is now obsolete, but harmless.

`MBEDTLS_PSA_INJECT_ENTROPY` should not be enabled by default for Mbed Crypto. It should only be enabled on targets that use this feature. In 5.12, it was automatically enabled on targets where `TARGET_PSA` and `MBEDTLS_ENTROPY_NV_SEED` were both enabled. It would have failed if such a target didn't also define `MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES`, so I must still be missing something, because I can't see any place where this gets defined.

Now `MBEDTLS_PSA_INJECT_ENTROPY` needs to be enabled explicitly on the targets that want this feature. I don't know which targets want it. This has to be a subset of the targets where it works.

To make this more backward compatible with 5.12, it would be possible to enable the feature where it's supported by modifying the configuration at the end of `features/mbedtls/platform/inc/platform_mbed.h`:
```
/* Enable entropy injection if it is supported. */
#if defined(MBEDTLS_ENTROPY_NV_SEED) && \
    defined(MBEDTLS_NO_DEFAULT_ENTROPY_SOURCES) && \
    defined(MBEDTLS_PSA_CRYPTO_STORAGE_C)
#define MBEDTLS_PSA_INJECT_ENTROPY
#endif
```
But I don't think this is desirable, because this feature adds a few hundred bytes of code and increases the attack surface and the risk of misconfiguration that could lead to bricking a device.

Reply:
I don't know the history of the aforementioned options beyond the fact that I have seen them being switched on and off in the past. However, based on this

> MBEDTLS_PSA_INJECT_ENTROPY should not be enabled by default for Mbed Crypto

which makes sense to me, why don't we just remove the macro we are not commenting on?
Right now I'm testing that approach locally.

Reply:
It builds with the feature switched off, but as far as I know, we actually need it in. Unfortunately, I don't know a simple way of enabling it without causing the trouble other than doing it for particular targets, as Gilles has mentioned earlier.

Reply:
@alzix @orenc17  Any suggestions ?

Reply:
I've come up with a short term fix in enabling the entropy injection API on PSA targets if the user or applications has chosen to enable NV Seed, only on PSA targets. This is compatible with Mbed OS 5.12.

Longer term, we need to implement entropy mixing in Mbed Crypto to support using both TRNG and NV Seed at the same time.

Reply:
I've tested on NUCLEO_F429ZI, K64F, and CY8CKIT_062_WIFI_BT.